### PR TITLE
Catch cryptic error when a plugin is not found

### PR DESCRIPTION
**Pull request by jfroffice**

### CODE REVIEW

**Comment by Atinux** [comment]
Related to #1778

File: lib/builder/builder.js [hunk 1]
```diff
@@ -62,8 +62,11 @@ export default class Builder extends Tapable {
   get plugins () {
     return this.options.plugins.map((p, i) => {
       if (typeof p === 'string') p = { src: p }
-      p.src = this.nuxt.resolvePath(p.src)
-      return { src: p.src, ssr: (p.ssr !== false), name: `plugin${i}` }
+      return {
+        src: this.nuxt.resolvePath(p.src),
+        ssr: (p.ssr !== false),
+        name: p.src.replace(/[^a-zA-Z?\d\s:]/g, '')
```
Comment (pi0) [inline lib/builder/builder.js]: Can't we wrap `p.src` into [path.basename](https://nodejs.org/api/path.html#path_path_basename_path_ext) ?

**Comment by jfroffice** [comment]
@pi0 

I am still getting crash on node 6.x

```
FATAL ERROR: CALL_AND_RETRY_LAST Allocation failed - JavaScript heap out of memory
 1: node::Abort() [/home/travis/.nvm/versions/node/v6.11.4/bin/node]
 2: 0x109f13c [/home/travis/.nvm/versions/node/v6.11.4/bin/node]
 3: v8::Utils::ReportApiFailure(char const*, char const*) [/home/travis/.nvm/versions/node/v6.11.4/bin/node]
 4: v8::internal::V8::FatalProcessOutOfMemory(char const*, bool) [/home/travis/.nvm/versions/node/v6.11.4/bin/node]
 5: v8::internal::Factory::NewTransitionArray(int) [/home/travis/.nvm/versions/node/v6.11.4/bin/node]
 6: v8::internal::TransitionArray::Insert(v8::internal::Handle<v8::internal::Map>, v8::internal::Handle<v8::internal::Name>, v8::internal::Handle<v8::internal::Map>, v8::internal::SimpleTransitionFlag) [/home/travis/.nvm/versions/node/v6.11.4/bin/node]
 7: v8::internal::Map::CopyReplaceDescriptors(v8::internal::Handle<v8::internal::Map>, v8::internal::Handle<v8::internal::DescriptorArray>, v8::internal::Handle<v8::internal::LayoutDescriptor>, v8::internal::TransitionFlag, v8::internal::MaybeHandle<v8::internal::Name>, char const*, v8::internal::SimpleTransitionFlag) [/home/travis/.nvm/versions/node/v6.11.4/bin/node]
 8: v8::internal::Map::CopyAddDescriptor(v8::internal::Handle<v8::internal::Map>, v8::internal::Descriptor*, v8::internal::TransitionFlag) [/home/travis/.nvm/versions/node/v6.11.4/bin/node]
 9: v8::internal::Map::CopyWithField(v8::internal::Handle<v8::internal::Map>, v8::internal::Handle<v8::internal::Name>, v8::internal::Handle<v8::internal::FieldType>, v8::internal::PropertyAttributes, v8::internal::Representation, v8::internal::TransitionFlag) [/home/travis/.nvm/versions/node/v6.11.4/bin/node]
10: v8::internal::Map::TransitionToDataProperty(v8::internal::Handle<v8::internal::Map>, v8::internal::Handle<v8::internal::Name>, v8::internal::Handle<v8::internal::Object>, v8::internal::PropertyAttributes, v8::internal::Object::StoreFromKeyed) [/home/travis/.nvm/versions/node/v6.11.4/bin/node]
11: v8::internal::LookupIterator::PrepareTransitionToDataProperty(v8::internal::Handle<v8::internal::JSObject>, v8::internal::Handle<v8::internal::Object>, v8::internal::PropertyAttributes, v8::internal::Object::StoreFromKeyed) [/home/travis/.nvm/versions/node/v6.11.4/bin/node]
12: v8::internal::Object::AddDataProperty(v8::internal::LookupIterator*, v8::internal::Handle<v8::internal::Object>, v8::internal::PropertyAttributes, v8::internal::Object::ShouldThrow, v8::internal::Object::StoreFromKeyed) [/home/travis/.nvm/versions/node/v6.11.4/bin/node]
13: v8::internal::Object::SetProperty(v8::internal::LookupIterator*, v8::internal::Handle<v8::internal::Object>, v8::internal::LanguageMode, v8::internal::Object::StoreFromKeyed) [/home/travis/.nvm/versions/node/v6.11.4/bin/node]
14: v8::internal::StoreIC::Store(v8::internal::Handle<v8::internal::Object>, v8::internal::Handle<v8::internal::Name>, v8::internal::Handle<v8::internal::Object>, v8::internal::Object::StoreFromKeyed) [/home/travis/.nvm/versions/node/v6.11.4/bin/node]
15: v8::internal::KeyedStoreIC::Store(v8::internal::Handle<v8::internal::Object>, v8::internal::Handle<v8::internal::Object>, v8::internal::Handle<v8::internal::Object>) [/home/travis/.nvm/versions/node/v6.11.4/bin/node]
16: v8::internal::Runtime_KeyedStoreIC_Miss(int, v8::internal::Object**, v8::internal::Isolate*) [/home/travis/.nvm/versions/node/v6.11.4/bin/node]
17: 0xe8a83b092a7
  ✖ test/ssr.test.js exited due to SIGABRT
```

**Review by pi0 — APPROVED** [review]
Seems working :)

**Comment by pi0** [comment]
Thanks, @jfroffice for this Enhancenment.

**Comment by pi0** [comment]
Also for possible memory leaks, we are making more investigations. That would be nice if you find out the exact reason and opening a PR for fixes.

**Pull request merged**